source/developers-guide/custom-shopping-world-elements/index.md

```diff
@@ -20,18 +20,24 @@ The shopping worlds are one highlight feature of Shopware. Here you can edit var
 <div class="toc-list"></div>
 
 ## Registering a new element ##
-For creating custom shopping world elements Shopware provides some helper functions which can be used in the `Bootstrap.php` of a [Shopware plugin](/developers-guide/plugin-quick-start/). So all you have to do is to create a simple plugin where you can register one ore more elements via the `createEmotionComponent()` method. As an example for this tutorial we will create a Vimeo element for adding videos to the shopping world.
+For creating custom shopping world elements Shopware provides some helper functions which can be used where the DependencyInjectionContainer is available. So all you have to do is to get the `shopware.emotion_component_installer` service in a simple plugin where you can register one ore more elements via the `createOrUpdate()` method. As an example for this tutorial we will create a Vimeo element for adding videos to the shopping world.
 
 ```php
-$vimeoElement = $this->createEmotionComponent([
-    'name' => 'Vimeo Video',
-    'xtype' => 'emotion-components-vimeo',
-    'template' => 'emotion_vimeo',
-    'cls' => 'emotion-vimeo-element',
-    'description' => 'A simple vimeo video element for the shopping worlds.'
-]);
+$this->emotionComponentInstaller = $this->container->get('shopware.emotion_component_installer');
+
+$vimeoElement = $this->emotionComponentInstaller->createOrUpdate(
+    $this->pluginName,
+    'SwagVimeoElement',
+    [
+        'name' => 'Vimeo Video',
+        'xtype' => 'emotion-components-vimeo',
+        'template' => 'emotion_vimeo',
+        'cls' => 'emotion-vimeo-element',
+        'description' => 'A simple vimeo video element for the shopping worlds.'
+    ]
+);
 ```
-In the `install()` method of our plugin we register a new element and save it in the variable `$vimeoElement` for later use. The `createEmotionComponent()` method expects a configuration array with the following properties:
+In the `install()` method of our plugin we register a new element. The `createOrUpdate()` method expects the plugin name, a element name and a configuration array with the following properties:
```

Comment:
you use a helper class in the install method which creates the new element

Reply:
D